fs/ftpfs.py

```diff
@@ -331,22 +345,28 @@ def ftp_url(self):
     @property
     def ftp(self):
         if self._ftp is None:
-            self._ftp = self._open_ftp()
+            self._ftp = self._open_ftp(self.encoding)
             self._welcome = self._ftp.getwelcome()
         return self._ftp
 
     @property
+    def encoding(self):
+        if self._ftp is None:
+            _ftp = self._open_ftp('latin-1')
+            return "utf-8" if "UTF8" in _ftp.sendcmd('FEAT') else "latin-1"
+        else:
+            return self._ftp.encoding
+
+    @property
     def features(self):
         """Get features dict from FTP server."""
         if self._features is None:
             try:
-                response = self.ftp.sendcmd(_encode("FEAT"))
+                response = _decode(self.ftp.sendcmd("FEAT"), "ascii")
```

Comment:
Is there a possibility of a unicode decode error here at all? I see the original did something similar, but it might be worth handling encoding problems for paranoia's sake.

Reply:
The [RFC](https://tools.ietf.org/html/rfc2389) does not say anything about encoding, but I believe features will only contain ASCII characters, so I guess there's no point over-interpreting the server response.

Reply:
I'd be tempted to handle the potential edge case where a response doesn't return 7-bit ascii. Maybe it won't ever happen, but there is a potential UnicodeDecodeError lurkng here.

How about `_decode` has unicode error handling set to `replace`? That way a flakey ftp server couldn't break this code.

Reply:
Well, if a server can't even get the features list right, I'm sure the code will break elsewhere at some point 😁 But I'll change the code accordingly.